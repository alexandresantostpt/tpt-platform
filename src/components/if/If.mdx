---
name: If
route: /if
---
import { Playground, PropsTable } from 'docz'

import If from './If'

import '@cfg/scripts'
import '@cfg/style'

# If

<PropsTable of={ If }/>

## Basic usage

```
<If condition={ BOOLEAN_VALUE } el={ <h2>Another JSX</h2> }>
    <h1>Must be show</h1>
</If>
```

<Playground>
    <If condition={ true } el={ <h2>Another JSX</h2> }>
        <h1>Must be show</h1>
    </If>
</Playground>
