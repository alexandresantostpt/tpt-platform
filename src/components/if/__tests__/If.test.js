import React from 'react'
import renderer from 'react-test-renderer'

import If from '@components/if/If'

describe('Tests for If component', () => {
    it('Should rendering without crash ', () => {
        const tree = renderer
            .create(
                <If condition={true}>
                    <h1>Must be show</h1>
                </If>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('Should rendering el props ', () => {
        const tree = renderer
            .create(
                <If condition={false} el={<h2>Must be show</h2>}>
                    <h1>Must not be show</h1>
                </If>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
