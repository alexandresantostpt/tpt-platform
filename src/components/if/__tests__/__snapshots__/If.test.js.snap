// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for If component Should rendering el props  1`] = `
<h2>
  Must be show
</h2>
`;

exports[`Tests for If component Should rendering without crash  1`] = `
<h1>
  Must be show
</h1>
`;
